Raycasting/src/Sound.cpp

```diff
@@ -0,0 +1,106 @@
+#include "Sound.h"
+
+Sound::Sound(const std::string & filepath) :
+	filepath(filepath)
+{
+}
+
+void Sound::setVolume(float volume)
+{
+	if (!isInitialized) {
+		sound.create(filepath.c_str());
+		sound.set2D(true);
+		isInitialized = true;
+	}
+	sound.setVolume(volume);
+}
+
+void Sound::setPlaybackSpeed(float speed)
+{
+	if (!isInitialized) {
+		sound.create(filepath.c_str());
+		sound.set2D(true);
+		isInitialized = true;
+	}
+	sound.setSpeed(speed);
+}
+
+void Sound::play2D()
+{
+	if (!isInitialized) {
+		sound.create(filepath.c_str());
+		sound.set2D(true);
+		isInitialized = true;
+	}
+	
+	sound.set2D(true);
+	sound.play();
+}
+
+void Sound::play3D(float x, float y, float z)
+{
+	if (!isInitialized) {
+		sound.create(filepath.c_str());
+		sound.set2D(true);
+		isInitialized = true;
+	}
+
+	is3D = true;
+	this->x = x;
+	this->y = y;
+	this->z = z;
+	makeSound();
+}
+
+void Sound::play3D(std::array<float, 2> observerPos, float observerFacingDirection, std::array<float, 3> soundPos)
```

Comment:
I'm not sure if the issue is here or not, but in the test program the explosion sound 'resets' after moving the mouse... its very peculiar....

Reply:
This is very peculiar... Might add to issues if I can't find it soon.

Reply:
I'm going to commit and push all the changes made so far before looking for a solution

Reply:
There were like a million errors!!! Who the hell let Sound pass review! Yall should be ashamed of yourselves!

Reply:
srry... 😢

Reply:
It has been rectified now, make sure you pull the most recent commit on this branch to get the fixes